plone/event/recurrence.py

```diff
@@ -103,8 +103,20 @@ def recurrence_sequence_ical(
             recrule = re.sub(r"T000000", t0str, recrule)
         # Then, replace each until times with the end of the day
         recrule = re.sub(
-            r"(UNTIL[^T]*[0-9]{8})T([0-9]{6}Z?)",
-            r"\1T235959",
+            r"(UNTIL[^T]*[0-9]{8})T([0-9]{6})(Z?)",
+            r"\1T235959\3",
+            recrule,
+        )
```

Comment:
I'm chewing a bit on keeping the UTC specifier here. IMO it makes more sense to always change to the end of the day in the local timezone of the event.

But anyways, AFAIK our widgets do not produce rrules with UNTIL dates localized to UTC.

So this should be fine.

I will add a comment.